editor/editor_help.cpp

```diff
@@ -798,7 +798,7 @@ void EditorHelp::_update_doc() {
 				class_desc->pop();
 				class_desc->push_font(doc_code_font);
 				String e = E->key();
-				if (e.get_slice_count(".")) {
+				if ((e.get_slice_count(".") > 1) && (e.get_slice(".", 0) == edited_class)) {
```

Comment:
Is this ever true? The only examples I know of are `Variant.Type` and `Variant.Operator`, as you mentioned they're exposed in `@GlobalScope` and not in the `Variant` docs, so even in the `Variant` doc they should stay referred to as `Variant.Type` and `Variant.Operator` (with `@GlobalScope` left out as implicit).

Reply:
I think there is a misunderstanding. This PR only changes the text of enum name in the enum definition section of documentation. Text of enum referrer links won't be affected. So if there is a `Variant.Type` link in the `Variant` page, the text is still "Variant.Type". By "omit the prefix if listed on the Variant page", I mean that if the definition is somehow moved from `@GlobalScope` to `Variant`, it should omit the prefix, just as the `Image.Format` enum is shown simplely as `Format` on the Image page.

The comparison is never true, currently. But it makes things more compatible. For example, the full name`Image.Format` is only used by referrers from other XMLs currently. If fullname is also used to document the enum in `Image.xml`, without the check, it'll be shown as `Image.Format` on the Image page, which is not ideal. I think this is exactly why there is a prefix-stripping process here previously. (Although it missed the `Variant.Type` on `@GlobalScope` page case, thus the issue.)

Reply:
Right, I misinterpreted the code and thought `e.get_slice_count(".") > 1` would only be true with two dots or more, but indeed if there's only one dot, that gives two slices.

Reply:
Right, I misinterpreted the code and thought `e.get_slice_count(".") > 1` would only be true with two dots or more, but indeed if there's only one dot, that gives two slices.